x-pack/plugins/lens/common/expressions/datatable/datatable.ts

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { cloneDeep } from 'lodash';
+import type {
+  DatatableColumnMeta,
+  ExpressionFunctionDefinition,
+} from '../../../../../../src/plugins/expressions/common';
+import type { FormatFactory, LensMultiTable } from '../../types';
+import type { ColumnConfigArg } from './datatable_column';
+import { getSortingCriteria } from './sorting';
+import { computeSummaryRowForColumn } from './summary';
+import { transposeTable } from './transpose_helpers';
+
+export interface DatatableProps {
+  data: LensMultiTable;
+  untransposedData?: LensMultiTable;
+  args: DatatableArgs;
+}
+
+export interface DatatableRender {
+  type: 'render';
+  as: 'lens_datatable_renderer';
+  value: DatatableProps;
+}
+
+export interface DatatableArgs {
+  title: string;
+  description?: string;
+  columns: ColumnConfigArg[];
+  sortingColumnId: string | undefined;
+  sortingDirection: 'asc' | 'desc' | 'none';
```

Comment:
This change LGTM, previously it was called `sorting?: SortingState` in `datatable_visualization/visualization.tsx`